api/server/v1/tx.go

```diff
@@ -92,6 +92,11 @@ func GetTransactionLegacy(req proto.Message) *TransactionCtx {
 		return r.GetTxCtx()
 	case *RollbackTransactionRequest:
 		return r.GetTxCtx()
+	case *PublishRequest:
+		if r.GetOptions() == nil || r.GetOptions().GetWriteOptions() == nil {
+			return nil
+		}
+		return r.GetOptions().GetWriteOptions().GetTxCtx()
```

Comment:
So this is old code that needs cleanup, we are not using TxCtx from the options, just remove it from here.